NumLib/DOF/DOFTableUtil.cpp

```diff
@@ -30,4 +44,60 @@ std::vector<GlobalIndexType> getIndices(
     return indices;
 }
 
+double norm(GlobalVector const& x, unsigned const global_component,
+            MathLib::VecNormType norm_type,
+            LocalToGlobalIndexMap const& dof_table, MeshLib::Mesh const& mesh)
+{
+    // TODO that also includes ghost nodes.
+    std::size_t const n_nodes = mesh.getNumberOfNodes();
+
+    // TODO for PETSc some global accumulation is necessary.
+    switch(norm_type)
+    {
+    case MathLib::VecNormType::NORM1:
```

Comment:
Currently different functions are not needed, since this function is only used in the convergence criteria #1342. The advantage of passing the norm type by parameter is that it is configurable at runtime without polymorphism.